doc/tutorials/cne/cne.txt

```diff
@@ -0,0 +1,356 @@
+network.Add<LogSoftMax<> >();
+@endcode
+
+first an object is created with the name network of FFN (feed forward network)
+
+User can add layers by calling the ADD method and specifying the type of 
+layer and the number of nodes in that particular layer and the next layer.
+
+In our example we will be using-
+2 input
+2 hidden and
+2 output layer
+
+
+@code
+network.Train(train, labels, opt);
+@endcode
+
+The train method take the following three parameters-
+
+1) train: The armadillo training data matrix.
+
+Note: Data points are arranged columnwise, where each column represents one
+      data point. Therefore the number of training data provided is the 
+      number of columns in the dataset.
+
+2) labels: The output of the training data in armadillo format.
+
+Note: This is also columnwise as the training dataset matrix.
+
+3) opt: The type of optimizer. We will be using CNE in this tutorial.
+
+
+@code
+network.Predict(test, predictions);
+@endcode
+
+The predict method can be called after training to obtain the result.
+The parameter definitions are-
+
+1) test: armadillo test set matrix in the above test set specified format.
+
+2) predictors: Will be modified by the model and output based on the test
+              case prediction will be added in this matrix.
+
+
+@subsection cne_ex3_cnetut Complete example
+
+In this example we will have two input nodes and the output is XOR of
+the two values. Where if node 1 gets activated means 0 as output and
+defines 2 means 1 as the XOR output
+As mentioned before we have 2 input, 2 hidded and 2 output node.
+
+@code
+#include <mlpack/core.hpp>
+
+#include <mlpack/methods/ann/layer/layer.hpp>
+#include <mlpack/methods/ann/ffn.hpp>
+
+#include <mlpack/core/optimizers/cne/cne.hpp>
+
+using namespace mlpack;
+using namespace mlpack::ann;
+using namespace mlpack::optimization;
+
+int main()
+{
+ /*
+   * Create the four cases for XOR with two variable
+   *
+   *  Input    Output
+   * 0 XOR 0  =  0
+   * 1 XOR 1  =  0 
+   * 0 XOR 1  =  1
+   * 1 XOR 0  =  1
+   */
+  arma::mat train("1,0,0,1;1,0,1,0");
+  arma::mat labels("1,1,2,2");
+
+  // network with 2 input 2 hidden and 2 output layer
+  FFN<NegativeLogLikelihood<> > network;
+
+  network.Add<Linear<> >(2, 2);
+  network.Add<SigmoidLayer<> >();
+  network.Add<Linear<> >(2, 2);
+  network.Add<LogSoftMax<> >();
+
+  // CNE object
+  CNE opt(20, 5000, 0.1, 0.02, 0.2, 0, 0);
+
+  // Training the network with CNE
+  network.Train(train, labels, opt);
+
+  // Predicting for the same train data
+  arma::mat predictionTemp;
+  network.Predict(train, predictionTemp);
+
+  arma::mat prediction = arma::zeros<arma::mat>(1, predictionTemp.n_cols);
+
+  for (size_t i = 0; i < predictionTemp.n_cols; ++i)
+  {
+    prediction(i) = arma::as_scalar(arma::find(
+        arma::max(predictionTemp.col(i)) == predictionTemp.col(i), 1)) + 1;
+  }
+
+  // Print the results 
+  for(size_t i = 0; i < 4; i++)
+    std::cout << prediction[i] << std::endl;
```

Comment:
You can also do `std::cout << prediction` here and there is no need for a loop. :)